ADAL/src/ADAuthenticationContext+Internal.m

```diff
@@ -83,8 +83,16 @@ + (ADAuthenticationError*)errorFromDictionary:(NSDictionary*)dictionary
         NSUUID* correlationId = [dictionary objectForKey:OAUTH2_CORRELATION_ID_RESPONSE] ?
                                 [[NSUUID alloc] initWithUUIDString:[dictionary objectForKey:OAUTH2_CORRELATION_ID_RESPONSE]]:
                                 nil;
-        SAFE_ARC_AUTORELEASE(correlationId);
-        return [ADAuthenticationError OAuthServerError:serverOAuth2Error description:errorDetails code:errorCode correlationId:correlationId];
+        SAFE_ARC_AUTORELEASE(correlationId);        
+        NSString *subcode = [dictionary objectForKey:OAUTH2_ERROR_SUBCODE];
```

Comment:
I'm going to have to check with the server team on this one. "error_subcode" isn't mentioned anywhere in the OAuth 2.0 spec (https://tools.ietf.org/html/rfc6749), so we'll have to make sure this behavior is intended and guaranteed.